src/wwwroot/app/services/CandidateService.js

```diff
@@ -1,32 +1,321 @@
+   new ThesaurusHelper('typeOfEmployment',  'typeOfEmployment',  'typeOfEmployment'),
+   new ThesaurusHelper('level',             'levelId',           'level'),
+   new ThesaurusHelper('currency',          'currencyId',        'currency'),
+   new ThesaurusHelper('socialNetwork',     'socialNetworks',    'socials')
+];
+
+let _HttpService, _ThesaurusService, _$q;
 
 export default class CandidateService {
-   constructor(HttpService) {
+   constructor(HttpService, ThesaurusService, $q) {
       'ngInject';
       _HttpService = HttpService;
+      _ThesaurusService = ThesaurusService;
+      _$q = $q;
    }
 
    getCandidates() {
-      return _HttpService.get(CANDIDATE_URL);
+      return _HttpService.get(CANDIDATE_URL).then(_forEach(_convertToClientFormat));
    }
 
    getCandidate(id) {
-      let additionalUrl = CANDIDATE_URL + id;
-      return _HttpService.get(additionalUrl);
+      return _HttpService.get(CANDIDATE_URL + id).then(_convertToClientFormat);
    }
 
    saveCandidate(entity) {
-      if (entity.Id) {
-         let additionalUrl = CANDIDATE_URL + entity.Id;
-         return _HttpService.put(additionalUrl, entity);
-      } else {
-         return _HttpService.post(CANDIDATE_URL, entity);
-      }
+      return _saveNewThesaurusTopics(entity)
+         .then(() => _convertToServerFormat(entity))
+         .then(candidate => {
+            if (candidate.id) {
+               return _HttpService.put(CANDIDATE_URL + candidate.id, candidate);
+            } else {
+               return _HttpService.post(CANDIDATE_URL, candidate);
+            }
+         })
+         .then(_convertToClientFormat)
+         .catch(() => {
+            _convertToClientFormat(entity);
+         });
    }
 
    deleteCandidate(entity) {
-      let additionalUrl = CANDIDATE_URL + entity.Id;
-      _HttpService.remove(additionalUrl, entity);
+      _HttpService.remove(CANDIDATE_URL + entity.id, entity);
+   }
+}
+
+function _addSavedThesaurusesTopics(candidate, thesaurusName, topics) {
+   candidate[thesaurusName] = union(candidate[thesaurusName] || [],  topics);
+   return candidate;
+}
+
+function _saveNewThesaurusTopics(candidate) {
+   let promises = reduce(THESAURUSES, (memo, helper) => {
+      let newTopicPattern = {id : undefined};
+      let topicsToSave = filter(candidate[helper.clientField], newTopicPattern);
+      if (topicsToSave.length) {
+         remove(candidate[helper.clientField], newTopicPattern);
+         memo[helper.clientField] = _ThesaurusService.saveThesaurusTopics(helper.thesaurusName, topicsToSave)
+            .then(_addSavedThesaurusesTopicsCurried(candidate, helper.thesaurusName));
+      }
+      return memo;
+   }, {});
+   return _$q.all(promises);
+}
+
+function  _convertThesaurusToIds(candidate) {
+   forEach(THESAURUSES, helper => {
+      if (helper.needConvertForServer) {
+         candidate[helper.serverField] = map(candidate[helper.clientField], topic => topic.id);
```

Comment:
check this
`candidate[helper.serverField] = map(candidate[helper.clientField], 'id')`